test/hooks.test.js

```diff
@@ -1,103 +1,120 @@
 import React, { useState, useEffect } from "react"
 import { observer, Observer, useLocalStore, useAsObservableSource } from "../src"
-import { sleepHelper } from "."
 import renderer, { act } from "react-test-renderer"
 
 test("computed properties react to props when using hooks", async () => {
-    const seen = []
+    jest.useFakeTimers()
 
-    const Child = ({ x }) => {
-        const props = useAsObservableSource({ x })
-        const store = useLocalStore(() => ({
-            get getPropX() {
-                return props.x
-            }
-        }))
+    try {
+        const seen = []
 
-        return <Observer>{() => (seen.push(store.getPropX), <div>{store.getPropX}</div>)}</Observer>
-    }
+        const Child = ({ x }) => {
+            const props = useAsObservableSource({ x })
+            const store = useLocalStore(() => ({
+                get getPropX() {
+                    return props.x
+                }
+            }))
+
+            return (
+                <Observer>
+                    {() => (seen.push(store.getPropX), <div>{store.getPropX}</div>)}
+                </Observer>
+            )
+        }
 
-    const Parent = () => {
-        const [state, setState] = useState({ x: 0 })
-        seen.push("parent")
-        useEffect(() => {
-            setTimeout(() => {
-                act(() => {
-                    setState({ x: 2 })
+        const Parent = () => {
+            const [state, setState] = useState({ x: 0 })
+            seen.push("parent")
+            useEffect(() => {
+                setTimeout(() => {
+                    act(() => {
+                        setState({ x: 2 })
+                    })
                 })
-            }, 100)
-        }, [])
-        return <Child x={state.x} />
-    }
+            }, [])
+            return <Child x={state.x} />
+        }
 
-    let wrapper
-    act(() => {
-        wrapper = renderer.create(<Parent />)
-    })
-    expect(wrapper.toJSON()).toMatchInlineSnapshot(`
+        let wrapper
+        act(() => {
+            wrapper = renderer.create(<Parent />)
+        })
+        expect(wrapper.toJSON()).toMatchInlineSnapshot(`
 <div>
   0
 </div>
 `)
 
-    await sleepHelper(400)
-    expect(seen).toEqual(["parent", 0, "parent", 2])
-    expect(wrapper.toJSON()).toMatchInlineSnapshot(`
+        jest.runAllTimers()
+
+        expect(seen).toEqual(["parent", 0, "parent", 2])
+        expect(wrapper.toJSON()).toMatchInlineSnapshot(`
 <div>
   2
 </div>
 `)
+    } finally {
```

Comment:
Does it really matter? I mean if a test fails then it's fine that timers are not restored because that dies with that failed Jest process. Or safer alternative can be to restore in `beforeEach`.

Reply:
> Does it really matter? I mean if a test fails then it's fine that timers are not restored because that dies with that failed Jest process.
I think that it is matter because without this if the first test fails, so `jest.useRealTimers` is not called, then the second test will fail too, although it should not. It'll be false negative.
> Or safer alternative can be to restore in `beforeEach`.
I've been thinking about it. I have some doubts about that. If we add a new test into this file in the future which won't need timers, `jest.useRealTimers` will be called anyway 🤔

Reply:
Calling `jest.useRealTimers` while fake ones are not used won't do a harm. It's a noop in that case.

Reply:
I don't have any more convincing arguments, so I'm gonna change that :)

Reply:
Done.

Reply:
Thanks! Much nicer I would say.

Reply:
@FredyC Thank you so much for the review! I hope I'm not boring you yet with my PRs? :)

Reply:
Well, tests are never the best excitement, but I am grateful you don't mind working on that :)

Reply:
Thanks!